### PR TITLE
Do not parse CSS on missed plugins or parser/stringifier option #1651

### PR DESCRIPTION
**Pull request by WilhelmYakunin**
Kindly let me know if I understood the purpose correctly. I see that some extra changes in tests files should be made.

### CODE REVIEW

File: lib/no-work.js [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+              add(plugin, event, plugin[event])
+            }
+          }
+        }
+      }
+    }
+    this.hasListener = Object.keys(this.listeners).length > 0
+  }
+
+  visitTick(stack) {
```
[the remark this answers is not in the document]
Comment (WilhelmYakunin) [reply lib/no-work.js]: Done. But I think there's a space for improvment

File: lib/document.js [hunk 2]
```diff
@@ -15,16 +15,32 @@ class Document extends Container {
...
   toResult(opts = {}) {
-    let lazy = new LazyResult(new Processor(), this, opts)
+    let result
+
+    if (
+      typeof opts.plugins === 'undefined' &&
+      typeof opts.parser === 'undefined' &&
+      typeof opts.stringifier === 'undefined' &&
+      typeof opts.syntax === 'undefined' &&
+      !opts.hideNothingWarning
```
Comment (ai) [inline lib/document.js:25]: We do not need `opts.hideNothingWarning`

File: lib/no-work.js [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+      typeof css === 'object' &&
+      css !== null &&
+      (css.type === 'root' || css.type === 'document')
+    ) {
+      root = cleanMarks(css)
+    } else {
+      let parser = parse
+      if (parser.parse) parser = parser.parse
+
+      root = parser(css, opts)
```
Comment (ai) [inline lib/no-work.js:36]: Why do you call `parser` if nobody touched `root` yet?

Comment (ai) [reply lib/no-work.js:36]: Seems like you didn’t understand the origin idea of this task

**Comment by ai** [comment]
I am closing this PR since it is not connected to origin task. Open PR will block other developers from trying to solve this task.

Feel free to reach me in Telegram `@sitnik` if you want a description what is this task about.

**Comment by WilhelmYakunin** [comment]
Ok

**Pull request closed without merging**